### PR TITLE
Enable local echoes for media uploads on development builds.

### PR DESCRIPTION
**Pull request by stefanceriu**
https://github.com/user-attachments/assets/e54fdccc-5e79-417c-9b1a-370f98309f1c

https://github.com/user-attachments/assets/a234f7f9-9179-4dee-9372-38f29d7b2a50

### CODE REVIEW

File: ElementX/Sources/Screens/MediaUploadPreviewScreen/MediaUploadPreviewScreenViewModel.swift [hunk 1]
```diff
@@ -75,29 +73,29 @@ class MediaUploadPreviewScreenViewModel: MediaUploadPreviewScreenViewModelType,
...
     }
     
     private static let loadingIndicatorIdentifier = "\(MediaUploadPreviewScreenViewModel.self)-Loading"
     
-    private func startLoading(progressPublisher: CurrentValuePublisher<Double, Never>) {
+    private func startLoading() {
         userIndicatorController.submitIndicator(
             UserIndicator(id: Self.loadingIndicatorIdentifier,
-                          type: .modal(progress: .published(progressPublisher), interactiveDismissDisabled: false, allowsInteraction: true),
+                          type: .modal(progress: .indeterminate, interactiveDismissDisabled: false, allowsInteraction: true),
```
Comment (pixlwave) [inline ElementX/Sources/Screens/MediaUploadPreviewScreen/MediaUploadPreviewScreenViewModel.swift:98]: Errr, can we not just do `AppSettings.isDevelopmentBuild ? .indeterminate : .published(progressPublisher)` here? Removing progress when uploading a video will be a regression for production users.

**Review by pixlwave — CHANGES_REQUESTED** [review]

File: ElementX/Sources/Screens/MediaUploadPreviewScreen/MediaUploadPreviewScreenViewModel.swift (hunk 1 above)
Comment (stefanceriu) [reply ElementX/Sources/Screens/MediaUploadPreviewScreen/MediaUploadPreviewScreenViewModel.swift:98]: I honestly didn't plan for it to stick around for that long, it's working just fine.

File: ElementX/Sources/Screens/Timeline/View/TimelineItemViews/ImageRoomTimelineView.swift [hunk 2]
```diff
@@ -44,6 +37,31 @@ struct ImageRoomTimelineView: View {
...
+                          size: timelineItem.content.thumbnailSize ?? timelineItem.content.size,
+                          mediaProvider: context.mediaProvider) {
+                placeholder
+            }
+            .timelineMediaFrame(height: timelineItem.content.thumbnailSize?.height ?? timelineItem.content.size?.height,
+                                aspectRatio: timelineItem.content.thumbnailAspectRatio ?? timelineItem.content.aspectRatio)
+        }
+    }
+    
     var source: MediaSourceProxy {
```
Comment (pixlwave) [inline ElementX/Sources/Screens/Timeline/View/TimelineItemViews/ImageRoomTimelineView.swift:65]: I don't think this is used now?

File: ElementX/Sources/Services/Timeline/TimelineItems/RoomTimelineItemFactory.swift [hunk 3]
```diff
@@ -513,29 +517,49 @@ struct RoomTimelineItemFactory: RoomTimelineItemFactoryProtocol {
...
+        var imageAspectRatio: CGFloat?
+        
+        if let imageInfo = messageContent.info,
+           let height = imageInfo.height,
+           let width = imageInfo.width {
+            imageSize = .init(width: CGFloat(width), height: CGFloat(height))
+            
+            if width > 0, height > 0 {
+                imageAspectRatio = CGFloat(width) / CGFloat(height)
+            }
```
Comment (pixlwave) [inline ElementX/Sources/Services/Timeline/TimelineItems/RoomTimelineItemFactory.swift:550]: Maybe we should pull this out into its own function that returns a tuple as we're using it 5 times.

**Review by pixlwave — APPROVED** [review]
LGTM 👍

File: fastlane/README.md
[the remark this answers is not in the document]
Comment (stefanceriu) [reply fastlane/README.md:1]: Ah, right 👍

**Pull request merged**